File: src/main/java/com/arcao/geocaching4locus/authentication/fragment/BasicMembershipWarningDialogFragment.java
package com.arcao.geocaching4locus.authentication.fragment;

import android.support.annotation.NonNull;
import com.afollestad.materialdialogs.DialogAction;
import com.afollestad.materialdialogs.MaterialDialog;
import com.arcao.geocaching4locus.R;
import com.arcao.geocaching4locus.base.fragment.AbstractErrorDialogFragment;
import com.arcao.geocaching4locus.base.constants.AppConstants;
import com.arcao.geocaching4locus.base.util.IntentUtil;

public class BasicMembershipWarningDialogFragment extends AbstractErrorDialogFragment {
  public static final String FRAGMENT_TAG = BasicMembershipWarningDialogFragment.class.getName();

  public static BasicMembershipWarningDialogFragment newInstance() {
    BasicMembershipWarningDialogFragment fragment = new BasicMembershipWarningDialogFragment();
    fragment.prepareDialog(R.string.title_basic_member_warning, R.string.warning_basic_member, null);
    return fragment;
  }

  @Override
  protected void onPositiveButtonClick() {
    super.onPositiveButtonClick();
    getActivity().finish();
    dismiss();
  }

  @Override
  protected void onDialogBuild(MaterialDialog.Builder builder) {
    super.onDialogBuild(builder);

    // disable auto dismiss
    builder.autoDismiss(false);

    builder.neutralText(R.string.button_show_manual)
        .onNeutral(new MaterialDialog.SingleButtonCallback() {
          @Override
          public void onClick(@NonNull MaterialDialog dialog, @NonNull DialogAction which) {
            IntentUtil.showWebPage(getActivity(), AppConstants.MANUAL_URI);
          }
        });
  }
}

File: app/src/main/java/com/arcao/geocaching4locus/authentication/fragment/BasicMembershipWarningDialogFragment.kt
package com.arcao.geocaching4locus.authentication.fragment

import com.afollestad.materialdialogs.MaterialDialog
import com.arcao.geocaching4locus.R
import com.arcao.geocaching4locus.base.constants.AppConstants
import com.arcao.geocaching4locus.base.fragment.AbstractErrorDialogFragment
import com.arcao.geocaching4locus.base.util.showWebPage

class BasicMembershipWarningDialogFragment : AbstractErrorDialogFragment() {

    override fun onPositiveButtonClick() {
        super.onPositiveButtonClick()
        requireActivity().finish()
        dismiss()
    }

    @Suppress("DEPRECATION")
    override fun onDialogBuild(dialog: MaterialDialog) {
        super.onDialogBuild(dialog)

        // disable auto dismiss
        dialog.noAutoDismiss()
            .neutralButton(R.string.button_show_users_guide) {
                requireActivity().showWebPage(AppConstants.USERS_GUIDE_URI)
            }
    }

    companion object {
        fun newInstance(): BasicMembershipWarningDialogFragment {
            val fragment = BasicMembershipWarningDialogFragment()
            fragment.prepareDialog(R.string.title_basic_member_warning, R.string.warning_basic_member, null)
            return fragment
        }
    }
}
